src/views/Playlist/PrivatePlaylist.vue

```diff
@@ -3,20 +3,26 @@ import {
   GetTrackCollectionModel,
   TrackCollectionApi,
 } from "@bcc-code/bmm-sdk-fetch";
-import { ref } from "vue";
+import { ref, watch } from "vue";
 
 const props = defineProps<{
   id: string;
 }>();
 
 const trackCollection = ref<GetTrackCollectionModel>({});
 
-new TrackCollectionApi()
-  .trackCollectionIdGet({ id: Number(props.id) })
-  .then((collection) => {
-    trackCollection.value = collection;
-  })
-  .catch(() => {});
+watch(
+  props,
```

Comment:
Same here, watch `props.id` instead of the whole props object:)

```ts
watch(
  () => props.id,
  newId => {
    new TrackCollectionApi()
      .trackCollectionIdGet({ id: Number(newId) })
  }
)
```